assets/js/state/selectors/runningOperations.js

```diff
@@ -1,20 +1,30 @@
 import { createSelector } from '@reduxjs/toolkit';
-import { find, flatMap, get } from 'lodash';
+import { pipe, find, castArray } from 'lodash/fp';
+import { flatMap, get } from 'lodash';
 
-const getRunningOperations = createSelector(
-  [(state) => state.runningOperations],
-  (runningOperations) => runningOperations
-);
+const getRunningOperations = ({ runningOperations }) => runningOperations;
 
 export const getRunningOperation = (groupID) =>
-  createSelector([getRunningOperations], (runningOperations) =>
-    get(runningOperations, groupID, {})
+  createSelector(getRunningOperations, (runningOperations) =>
+    get(runningOperations, groupID, null)
   );
 
-export const getRunningOperationsList = () =>
-  createSelector([getRunningOperations], (runningOperations) =>
+export const getRunningOperationsList = createSelector(
+  getRunningOperations,
+  (runningOperations) =>
     flatMap(runningOperations, (value, key) => ({ groupID: key, ...value }))
-  );
+);
+
+const defaultMatcher = (_) => true;
 
-export const isOperationRunning = (runningOperationsList, groupID, operation) =>
-  !!find(runningOperationsList, { groupID, operation });
+export const isOperationRunning = (
+  runningOperationsList,
+  groupID,
+  operation,
+  matcher = defaultMatcher
+) =>
+  pipe(
+    castArray,
```

Comment:
I don't think we should have this `castArray`. The function expects a list, otherwise the param name above is totally missleading.

Reply:
Would it be enough to change the param from `runningOperationsList` to something like `runningOperationItemOrList`?

Reply:
It wouldn't be harmful, but here it is 772726d2a90e2506bc1d6a655370c7132e5e8b1a